rocks-bin/src/outdated.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::HashMap;
+
+use clap::Args;
+use eyre::{OptionExt as _, Result};
+use itertools::Itertools;
+use rocks_lib::{
+    config::Config,
+    manifest::{manifest_from_server, ManifestMetadata},
+    tree::Tree,
+};
+use text_trees::{FormatCharacters, StringTreeNode, TreeFormatting};
+
+#[derive(Args)]
+pub struct Outdated {
+    #[arg(long)]
+    porcelain: bool,
+}
+
+pub async fn outdated(outdated_data: Outdated, config: &Config) -> Result<()> {
+    let tree = Tree::new(
+        &config.tree,
+        config
+            .lua_version
+            .as_ref()
+            .ok_or_eyre("lua version not supplied!")?,
```

Comment:
suggestion: extract to a method for `Config`.

Reply:
Do you mean a getter that returns an Option<&T>? If so, I agree.

Reply:
Note: on second glance, there is some benefit to allowing the configuration to be modified at runtime, so I don't think we should provide read-only fields for the config. Thoughts?

Reply:
> Do you mean a getter that returns an Option<&T>? If so, I agree.

I meant a getter that returns a `Result`.

> Note: on second glance, there is some benefit to allowing the configuration to be modified at runtime, so I don't think we should provide read-only fields for the config. Thoughts?

Do you have anything concrete in mind? If not, I'd say yagni for now.

Reply:
I was thinking that project detection could dynamically alter the `--tree` flag, but on second glance it can actually be done ahead of time in the default configuration, so you're right, YAGNI. However, I think that's fit for a separate PR, so I'll add a TODO comment and deal with that there :)